### PR TITLE
Refactor and cleanup

### PR DESCRIPTION
**Pull request by roryokane**
See individual commit messages for details.

I made this PR because I was reading the code to understand the algorithm better, and saw some ways to make the code clearer.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
 const INDENT_REGEX = /^(?:( )+|\t+)/;
 
 function getMostUsed(indents) {
-	let result = 0;
+	let result = null;
```
Comment (sindresorhus) [inline index.js]: I prefer just using `undefined`.
```suggestion
	let result;
```

File: index.js [hunk 2]
```diff
@@ -98,24 +82,68 @@ module.exports = string => {
...
-	const result = getMostUsed(indents);
+	return indents;
+}
 
-	let amount = 0;
-	let type;
-	let indent = '';
+// Encode the indent type and amount as a string (e.g. 's4') for use as a compound key in the indents Map.
+function encodeIndentsKey(indentType, indentAmount) {
+	const typeChar = indentType === INDENT_TYPE_SPACE ? 's' : 't';
```
Comment (sindresorhus) [inline index.js]:
```suggestion
	const typeCharacter = indentType === INDENT_TYPE_SPACE ? 's' : 't';
```

File: index.js [hunk 3]
```diff
@@ -98,24 +82,68 @@ module.exports = string => {
...
+	return result;
+}
+
+function makeIndentString(type, amount) {
+	if (type === INDENT_TYPE_SPACE) {
+		return ' '.repeat(amount);
+	}
+
+	return '\t'.repeat(amount);
+}
```
Comment (sindresorhus) [inline index.js]:
```js
function makeIndentString(type, amount) {
	const indent = type === INDENT_TYPE_SPACE ? ' ' : '\t';
	return indent.repeat(amount);
}
```

**Comment by sindresorhus** [comment]
Thanks for making the code more readable 🙌

**Comment by roryokane** [comment]
Feedback addressed in separate commits. If you want, I can squash the commits after you approve the PR.

**Comment by sindresorhus** [comment]
Perfect :)

**Pull request merged**